openproject/client.py

```diff
@@ -0,0 +1,131 @@
+import httpx
+from exceptions import APIError, AuthenticationError
+
+
+class Client:
+    def __init__(self, base_url: str, api_token: str):
+        self.base_url = base_url
+        self.api_version = "v3"
+        self.api_token = api_token
+
+    def _handle_response(self, response: httpx.Response):
+        if response.status_code == 401:
+            raise AuthenticationError(response.json()["message"], response.status_code)
+
+        if response.status_code == 204:
+            return None
+
+        try:
+            response.raise_for_status()
+
+            return response.json()
+        except:
+            error = f"{response.text}"
+            if "application/json" in response.headers["Content-Type"]:
+                resp = response.json()
+                if "message" in resp:
+                    error = f"Message: {resp.get('message')} , Error details: {resp.get('errors')}, {resp.get('data')}"
+                    raise APIError(error, response)
+
+                elif "msg" in resp:
+                    error = f"Message: {resp.get('msg')} , Error details: {resp.get('errors')}, {resp.get('data')}"
+                    raise APIError(error, response)
```

Comment:
So, I read a bit about how we can handle this in a more Pythonic way. Found something interesting.

@phemmylintry can you try the following to see if it works?

```py
error_message_keys = ("message", "msg")
error_message = next((resp[key] for key in error_message_keys if key in resp), None)
if error_message:
    raise ...
```

Reply:
Alright